File: backend/src/YouG.Application/Features/Groups/Dtos/JoinGroupResultDto.cs
namespace YouG.Application.Features.Groups.Dtos;

/// <summary>
/// `Joined` is true and `Group` populated when the invite led to instant membership (admin-
/// created link, or the user was already a member). When the invite was created by a non-admin
/// member, `Joined` is false and a `GroupJoinRequest` was created/reused instead, pending admin approval.
/// </summary>
public record JoinGroupResultDto(bool Joined, GroupDto? Group);

File: mobile/lib/features/groups/data/dtos/join_group_result_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'group_dto.dart';

part 'join_group_result_dto.freezed.dart';
part 'join_group_result_dto.g.dart';

/// Mirrors backend `JoinGroupResultDto` exactly. `group` is null when `joined` is false —
/// the invite led to a pending `GroupJoinRequest` instead of instant membership.
@freezed
abstract class JoinGroupResultDto with _$JoinGroupResultDto {
  const factory JoinGroupResultDto({required bool joined, GroupDto? group}) = _JoinGroupResultDto;

  factory JoinGroupResultDto.fromJson(Map<String, dynamic> json) => _$JoinGroupResultDtoFromJson(json);
}
